diff --git a/AdMipMapConfig.cmake.in b/AdMipMapConfig.cmake.in
new file mode 100644
index 0000000..af8e17c
--- /dev/null
+++ b/AdMipMapConfig.cmake.in
@@ -0,0 +1,4 @@
+@PACKAGE_INIT@
+
+include("${CMAKE_CURRENT_LIST_DIR}/AdMipMapTargets.cmake")
+check_required_components("@PROJECT_NAME@")
diff --git a/CMakeLists.txt b/CMakeLists.txt
new file mode 100644
index 0000000..969c941
--- /dev/null
+++ b/CMakeLists.txt
@@ -0,0 +1,44 @@
+cmake_minimum_required(VERSION 3.21)
+
+project(AdMipMap)
+
+set(SOURCES "${CMAKE_CURRENT_LIST_DIR}/ad_mipmap.h")
+
+add_library(AdMipMap INTERFACE ${SOURCES})
+add_library(AdMipMap::AdMipMap ALIAS AdMipMap)
+set_target_properties(AdMipMap PROPERTIES C_STANDARD 11)
+
+target_include_directories(AdMipMap INTERFACE
+    $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>
+    $<INSTALL_INTERFACE:include>
+)
+
+include(CMakePackageConfigHelpers)
+write_basic_package_version_file(
+    "${PROJECT_BINARY_DIR}/AdMipMapConfigVersion.cmake"
+    VERSION 1.0.0
+    COMPATIBILITY AnyNewerVersion
+)
+
+install(TARGETS AdMipMap
+    EXPORT AdMipMapTargets
+    LIBRARY DESTINATION lib COMPONENT Runtime
+    ARCHIVE DESTINATION lib COMPONENT Development
+    RUNTIME DESTINATION bin COMPONENT Runtime
+    PUBLIC_HEADER DESTINATION include COMPONENT Development
+    BUNDLE DESTINATION bin COMPONENT Runtime
+)
+
+include(CMakePackageConfigHelpers)
+configure_package_config_file(
+    "${CMAKE_CURRENT_LIST_DIR}/AdMipMapConfig.cmake.in"
+    "${PROJECT_BINARY_DIR}/AdMipMapConfig.cmake"
+    INSTALL_DESTINATION share/AdMipMap
+)
+
+install(EXPORT AdMipMapTargets DESTINATION share/AdMipMap)
+install(FILES "${PROJECT_BINARY_DIR}/AdMipMapConfigVersion.cmake"
+              "${PROJECT_BINARY_DIR}/AdMipMapConfig.cmake"
+        DESTINATION share/AdMipMap)
+
+install(FILES ${SOURCES} DESTINATION include)
